### ISSUE TITLE
USB discovery integration failure: denied «/sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.5/bNumInterfaces»

### ISSUE THREAD

**Issue by alabamatoy** [home-assistant-snap/home-assistant-snap#4]
Ubuntu 20.04 on Pi 3

```
Sep 28 14:15:24 ubuntu kernel: [   89.946905] kauditd_printk_skb: 32 callbacks suppressed
Sep 28 14:15:24 ubuntu kernel: [   89.946917] audit: type=1400 audit(1632838524.482:44): apparmor="DENIED" operation="open" profile="snap.home-assistant-snap.home-assistant-snap" name="/sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.5/bNumInterfaces" pid=1923 comm="python3" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]: #033[31m2021-09-28 14:15:24 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]: Traceback (most recent call last):
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:   File "/snap/home-assistant-snap/399/lib/python3.8/site-packages/homeassistant/components/usb/__init__.py", line 108, in async_start
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:     await self._async_scan_serial()
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:   File "/snap/home-assistant-snap/399/lib/python3.8/site-packages/homeassistant/components/usb/__init__.py", line 202, in _async_scan_serial
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:     self._async_process_ports(await self.hass.async_add_executor_job(comports))
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:   File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:     result = self.fn(*self.args, **self.kwargs)
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:   File "/snap/home-assistant-snap/399/lib/python3.8/site-packages/serial/tools/list_ports_linux.py", line 102, in comports
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:     for info in [SysFS(d) for d in devices]
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:   File "/snap/home-assistant-snap/399/lib/python3.8/site-packages/serial/tools/list_ports_linux.py", line 102, in <listcomp>
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:     for info in [SysFS(d) for d in devices]
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:   File "/snap/home-assistant-snap/399/lib/python3.8/site-packages/serial/tools/list_ports_linux.py", line 48, in __init__
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]:     num_if = int(self.read_line(self.usb_device_path, 'bNumInterfaces'))
Sep 28 14:15:24 ubuntu home-assistant-snap.home-assistant-snap[1923]: TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'#033[0m
```

**Comment by alabamatoy** [comment]
Sorry, code markup doesnt seem to work...

**Comment by jmgiaever** [comment]
Hi, @alabamatoy!

I updated the formatting for you. (you need three ticks ` for multiline formatting).

I'm aware of this. It's not kernel errors, but an issue with the new **USB discovery integration**. 

I've honestly not had time to dig into it, but it shouldn't affect anything else than the discovery of new USB devices you might add to you Pi. I haven't given it high priority since I assume most users knows what they have connected. 

Let me know if it's breaking anything in your setup. For what I'm aware you can just ignore it as long as you're not depending on the USB discovery integration?

**Comment by alabamatoy** [comment]
OK.  Just trying to make sure it wasn't something new.  Close at your discretion.  Thanks for what you do!

**Comment by jmgiaever** [comment]
I keep it open, so I have a reminder to look into it :)

**Comment by JohnMcLear** [comment]
On startup I get

```
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]: 2021-11-19 09:51:33 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]: Traceback (most recent call last):
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:   File "/snap/home-assistant-snap/423/lib/python3.8/site-packages/homeassistant/components/usb/__init__.py", line 108, in async_start
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:     await self._async_scan_serial()
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:   File "/snap/home-assistant-snap/423/lib/python3.8/site-packages/homeassistant/components/usb/__init__.py", line 217, in _async_scan_serial
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:     self._async_process_ports(await self.hass.async_add_executor_job(comports))
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:   File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:     result = self.fn(*self.args, **self.kwargs)
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:   File "/snap/home-assistant-snap/423/lib/python3.8/site-packages/serial/tools/list_ports_linux.py", line 102, in comports
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:     for info in [SysFS(d) for d in devices]
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:   File "/snap/home-assistant-snap/423/lib/python3.8/site-packages/serial/tools/list_ports_linux.py", line 102, in <listcomp>
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:     for info in [SysFS(d) for d in devices]
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:   File "/snap/home-assistant-snap/423/lib/python3.8/site-packages/serial/tools/list_ports_linux.py", line 48, in __init__
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]:     num_if = int(self.read_line(self.usb_device_path, 'bNumInterfaces'))
2021-11-19T09:51:33Z home-assistant-snap.home-assistant-snap[458834]: TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```
```
nap list home-assistant-snap 
Name                 Version    Rev  Tracking        Publisher       Notes
home-assistant-snap  2021.10.7  423  2021.10/stable  giaever-online  -
```
I switched to ``edge``

I tried 
```
snap connect some-snap-name:removable-media
```

```
# sudo snap connections home-assistant-snap 
Interface                Plug                                         Slot                                Notes
bluez                    home-assistant-snap:bluez                    -                                   -
content                  -                                            home-assistant-snap:components      -
content                  -                                            home-assistant-snap:configurations  -
content                  -                                            home-assistant-snap:vscs-content    -
content                  home-assistant-snap:bin                      -                                   -
desktop                  home-assistant-snap:desktop                  :desktop                            -
network                  home-assistant-snap:network                  :network                            -
network-bind             home-assistant-snap:network-bind             :network-bind                       -
network-control          home-assistant-snap:network-control          :network-control                    -
physical-memory-control  home-assistant-snap:physical-memory-control  -                                   -
raw-usb                  home-assistant-snap:raw-usb                  -                                   -
removable-media          home-assistant-snap:removable-media          :removable-media                    manual
serial-port              home-assistant-snap:serial-port              :usbserial                          manual
```

**Comment by jmgiaever** [comment]
Yeah, it's not fixed yet.

**Comment by jmgiaever** [comment]
It shouldn't have an huge impact. It's just USB discovery, so HA can discover whatever you've plugged in. But I haven't put a priority to it, since most users know what they have plugged in and do not need HA to discover it.

**Comment by jmgiaever** [comment]
Let me know if it breaks anything

**Comment by jmgiaever** [comment]
Hi, this should be fixed in `2021.12/stable` and `latest/stable` channels. You need the `hardware-observe` plug, `sudo snap connect home-assistant-snap hardware-observe`.

If you're on a different channel (check with `sudo snap info home-assistant-snap | grep tracking`) you can switch with 

```
sudo snap switch home-assistant-snap --channel=2021.12/stable
sudo snap refresh home-assistant-snap
```
